sstables/sstables.cc

```diff
@@ -1545,12 +1545,12 @@ populate_statistics_offsets(sstable_version_types v, statistics& s) {
 
 static
 sharding_metadata
-create_sharding_metadata(schema_ptr schema, const dht::decorated_key& first_key, const dht::decorated_key& last_key, shard_id shard) {
+create_sharding_metadata(schema_ptr schema, const dht::sharder& sharder, const dht::decorated_key& first_key, const dht::decorated_key& last_key, shard_id shard) {
     auto prange = dht::partition_range::make(dht::ring_position(first_key), dht::ring_position(last_key));
     auto sm = sharding_metadata();
```

Comment:
This is overkill for tablets+compaction groups.

When we have both tablets and compaction groups, it's enough to set sharding metadata = [first key, last key]. This is because, even if the sstable ends up spanning 2^n tablets with n > 0, there will never be "holes" that we need to ignore like the current sharding metadata.

So, if we end up throwing away the tablet sharder, this can be replaced with a bool_class\<generate_fine_grained_sharding_metadata> or whatever (which the table can propagate down).

Reply:
Yes, but this is necessary for incremental development. The node will not boot without this. Later, when sstables will span only a single tablet, this metadata will naturally reduce to [first key, last key].